src/test/kotlin/com/atlassian/performance/tools/ssh/api/SshTest.kt

```diff
@@ -56,14 +56,24 @@ class SshTest {
         SshContainer().useSsh { sshHost ->
             installPing(sshHost)
 
-            val fail = sshHost.runInBackground("nonexistent-command")
-            val failResult = fail.stop(Duration.ofMillis(20))
+            repeat(1000) {
+                val fail = sshHost.runInBackground("nonexistant-command")
+                sshHost.waitForAllProcessesToFinish("nonexistant-command", Duration.ofMillis(100))
+                val failResult = fail.stop(Duration.ofMillis(20))
 
-            Assert.assertEquals(127, failResult.exitStatus)
+                Assert.assertEquals(127, failResult.exitStatus)
+            }
         }
     }
 
     private fun installPing(sshHost: Ssh) {
         sshHost.newConnection().use { it.execute("apt-get update -qq && apt-get install iputils-ping -y") }
     }
+
+    private fun Ssh.waitForAllProcessesToFinish(
+        processCommand: String,
+        timeout: Duration
+    ) = this.newConnection().use {
+        it.execute("wait `pgrep '$processCommand'`", timeout)
```

Comment:
Probably a bit of an overkill, but I don't have a better idea how we can wait for the process to finish.

It's also a shotgun, because we are waiting not only for the one process we started, but also all of other processes that might be named the same. I think we don't expect there to be any other processes like that anyway.

Reply:
I think the chance of other processes with the same name is minimal, and this solution should be sufficient.